src/main/java/com/google/sps/data/DataStoreManager.java

```diff
@@ -0,0 +1,342 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import java.io.IOException;
+import java.lang.reflect.Type;
+import com.google.gson.reflect.TypeToken;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet responsible for creating new tasks. **/
+public class DatastoreManager implements DataManager {
+
+  private DatastoreService datastore = DatastoreServiceFactory.getDataStoreService();
+  Gson gson = new Gson();
+  
+  public DataStoreManager() {  
+  }
+
+  /**
+    * Stores the static data of a single game in datastore as an entity
+    * @param game an Game variable representing a single instance of a game.
+  */
+  public void storeGame(Game game) {
+    Entity gameEntity = new Entity("Game", game.getGameID());
+    gameEntity.setProperty("gameName", game.getGameName());
+    gameEntity.setProperty("gameCreator", game.getGameCreator());
+    gameEntity.setProperty("gameDescription", game.getGameDescription());
+    gameEntity.setProperty("numStages", game.getNumStages());
+    gameEntity.setProperty("stages", game.getStages());
+    gameEntity.setProperty("numTimesPlayed", game.getNumTimesPlayed());
```

Comment:
The method "storeGame" and "updateGame" are almost the same. It seems like the "updateGame" does the exactly same thing as the "storeGame", except retrieving the Game from the datastore.

I think we could make the "updateGame" to have empty fields. When users firstly create a game, we store the Game into the datastore using the "storeGame" API. When users update a game, we can only pass those updated information to the datastore. So some fields might be empty. We only update the non-empty fields to the datastore. 

So in the updateGame(), you can have something like

String gameName = game.getGameName();
if (gameName != null) {
    gameEntity.setProperty("gameName", game.getGameName());
}

But please check with Lawrence and Sara if they would like to have two separate APIs: storeGame and updateGame, or a single API storeGame.